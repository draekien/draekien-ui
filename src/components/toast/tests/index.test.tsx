/** @jsx jsx */
import { fireEvent, getByText, render } from '@testing-library/react';
import { jsx } from 'theme-ui';
import Toast from '..';

describe('Toast component', () => {
  test('should match snapshot', () => {
    const { container } = render(<Toast title="test" message="test" />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when information', () => {
    const { container } = render(
      <Toast color="information" title="test" message="test" openFrom="left" />
    );

    expect(container).toMatchSnapshot();
  });

  test('onClick successfully calls callback function', (done) => {
    const fn = jest.fn(() => done());
    const { container } = render(<Toast onClick={fn} title="test" />);

    fireEvent(
      getByText(container as HTMLElement, 'test'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );
  });

  test('onClose successfully calls callback function', (done) => {
    const fn = jest.fn(() => done());
    const { container } = render(<Toast onClose={fn} title="test" />);

    fireEvent(
      getByText(container as HTMLElement, 'close'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );
  });
});
