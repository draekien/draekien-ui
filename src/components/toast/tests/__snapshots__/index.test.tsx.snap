// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component should match snapshot 1`] = `
<div>
  <div
    class="css-wwl213"
    style="height: 0px;"
  >
    <i
      class="material-icons css-148htax-Icon"
    >
      check_circle
    </i>
    <div
      class="css-d3ud5j"
    >
      test
    </div>
    <div
      class="css-1o9fee3"
    >
      test
    </div>
    <i
      class="material-icons css-b6hqi9-Icon"
    >
      close
    </i>
  </div>
</div>
`;

exports[`Toast component should match snapshot when information 1`] = `
<div>
  <div
    class="css-1f0v0hv"
    style="height: 0px;"
  >
    <i
      class="material-icons css-150jhr4-Icon"
    >
      info
    </i>
    <div
      class="css-klbf7x"
    >
      test
    </div>
    <div
      class="css-1o9fee3"
    >
      test
    </div>
    <i
      class="material-icons css-b6hqi9-Icon"
    >
      close
    </i>
  </div>
</div>
`;
